src/main/java/com/direwolf20/buildinggadgets/client/gui/base/EntryList.java

```diff
@@ -0,0 +1,197 @@
+            bufferbuilder.pos(x2, l1, 0.0D).color(128, 128, 128, 255).endVertex();
+            bufferbuilder.pos(x1, l1, 0.0D).color(128, 128, 128, 255).endVertex();
+
+            bufferbuilder.pos(x1, (l1 + k1 - 1), 0.0D).color(192, 192, 192, 255).endVertex();
+            bufferbuilder.pos((x2 - 1), (l1 + k1 - 1), 0.0D).color(192, 192, 192, 255).endVertex();
+            bufferbuilder.pos((x2 - 1), l1, 0.0D).color(192, 192, 192, 255).endVertex();
+            bufferbuilder.pos(x1, l1, 0.0D).color(192, 192, 192, 255).endVertex();
+            tessellator.draw();
+        }
+
+        renderDecorations(mouseX, mouseX);
+        GlStateManager.enableTexture();
+        GlStateManager.shadeModel(GL_FLAT);
+        GlStateManager.enableAlphaTest();
+        GlStateManager.disableBlend();
+    }
+
+    protected void renderTransition(Tessellator tessellator, BufferBuilder buffer, int transitionTop, int transitionBottom) {
+        GlStateManager.enableBlend();
+        GlStateManager.blendFuncSeparate(SourceFactor.SRC_ALPHA, DestFactor.ONE_MINUS_SRC_ALPHA, SourceFactor.ZERO, DestFactor.ONE);
+        GlStateManager.disableAlphaTest();
+        GlStateManager.shadeModel(GL_SMOOTH);
+        buffer.begin(GL_QUADS, DefaultVertexFormats.POSITION_COLOR);
+        buffer.pos(x0, transitionTop, 0.0D).color(0, 0, 0, 255).endVertex();
+        buffer.pos(x1, transitionTop, 0.0D).color(0, 0, 0, 255).endVertex();
+        buffer.pos(x1, transitionBottom, 0.0D).color(0, 0, 0, 0).endVertex();
+        buffer.pos(x0, transitionBottom, 0.0D).color(0, 0, 0, 0).endVertex();
+        tessellator.draw();
+    }
+
+    protected void renderContentBackground(Tessellator tessellator, BufferBuilder bufferbuilder) {
+        fillGradient(getLeft(), getTop(), getRight(), getBottom(), 0xC0101010, 0xD0101010);
+    }
+
+    protected final void renderDefaultContentBackground(Tessellator tessellator, BufferBuilder bufferbuilder) {
+        this.minecraft.getTextureManager().bindTexture(AbstractGui.BACKGROUND_LOCATION);
+        GlStateManager.color4f(1.0F, 1.0F, 1.0F, 1.0F);
+        bufferbuilder.begin(GL_QUADS, DefaultVertexFormats.POSITION_TEX_COLOR);
+        float v1 = (float) (y1 + (int) getScrollAmount()) / 32.0F;
+        float v2 = (float) (y0 + (int) getScrollAmount()) / 32.0F;
+        float u1 = (float) x0 / 32.0F;
+        float u2 = (float) x1 / 32.0F;
+        bufferbuilder.pos(x0, y1, 0.0D).tex(u1, v1).color(32, 32, 32, 255).endVertex();
+        bufferbuilder.pos(x1, y1, 0.0D).tex(u2, v1).color(32, 32, 32, 255).endVertex();
+        bufferbuilder.pos(x1, y0, 0.0D).tex(u2, v2).color(32, 32, 32, 255).endVertex();
+        bufferbuilder.pos(x0, y0, 0.0D).tex(u1, v2).color(32, 32, 32, 255).endVertex();
+        tessellator.draw();
+    }
+
+    // No dirt top/bottom background
+    @Override
+    protected void renderHoleBackground(int p_renderHoleBackground_1_, int p_renderHoleBackground_2_, int p_renderHoleBackground_3_, int p_renderHoleBackground_4_) {
```

Comment:
When possible (in general) try and infer the method argument names when you can.

Reply:
Maybe something like:(?) 
```suggestion
    protected void renderNonDefaultBackground(int p_renderHoleBackground_1_, int p_renderHoleBackground_2_, int p_renderHoleBackground_3_, int p_renderHoleBackground_4_) {
```
I'm really unsure what a better name could be in this case - I don't like the current one though either. No idea what could be a better name though

Reply:
This "hole background" thing is the top/bottom of dirt background for AbstractList. Not sure why does it have a weird name, but it definitely can't exist in the current design

Reply:
I was referring to the method arg names and in a very general sense. Effectively, if you use a method with crap arg names then try and infer their actual properties.